### PR TITLE
fix(router): make relativeLinkResolution corrected by default

### LINKED ISSUE #37355: relativeLinkResolution: 'corrected' should be the default option
https://angular.io/api/router/ExtraOptions#relativeLinkResolution
`relativeLinkResolution: 'corrected` Enables a bug fix that corrects relative link resolution in components with empty paths. 
The following issues are fixed, but continue to cause user pain because the default option is 'legacy' (used when `undefined`): #19290, #27674, #18059

### PR DESCRIPTION
**Pull request by dirkluijk**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[X] Bugfix
```

## What is the current behavior?
To apply the fix of https://github.com/angular/angular/pull/22394 in Angular 6.x, you have to set `relativeLinkResolution` to  `corrected`.

## What is the new behavior?
This PR makes this the default for Angular 7.

## Does this PR introduce a breaking change?
```
[X] Yes
[ ] No
```
Routing behaviour will be affected for empty path routes.

## Other information
See https://github.com/angular/angular/pull/22394/files.

Credits to @adriensamson.

### CODE REVIEW

**Comment by alfaproject** [comment]
You need to deprecate first and then remove in v8

**Comment by dirkluijk** [comment]
@alfaproject Good point. Done.

**Comment by dirkluijk** [comment]
@jasonaden This PR did miss the 7.0 release. Can we make `corrected` default in Angular 8?

By the way, the integrations tests in this PR are now included in https://github.com/angular/angular/commit/a752971bca0ab434971b61569545b55d4c883505.

**Comment by sarunint** [comment]
What's the plan for this? In my opinion:

1. Change this to `corrected` by default in v8.
2. Deprecate this in v8, and remove it in v10?

**Comment by dirkluijk** [comment]
Yes please. 👍

File: packages/router/src/router.ts [hunk 1]
```diff
@@ -299,8 +299,10 @@ export class Router {
 
   /**
    * See {@link RouterModule} for more information.
+   *
+   * @deprecated since 7.0.0, will be removed in a later version
```
Comment (atscott) [inline packages/router/src/router.ts:303]: You'll also need an entry in `aio/content/guide/deprecations.md` for this deprecation.

File: packages/router/src/router_module.ts [hunk 2]
```diff
@@ -443,8 +443,9 @@ export interface ExtraOptions {
    *
    * `<a [routerLink]="['../a']">Link to A</a>`
    *
-   * In other words, you're required to use `../` rather than `./`. The current default in v6
-   * is `legacy`, and this option will be removed in v7 to default to the corrected behavior.
+   * In other words, you're required to use `../` rather than `./`.
+   *
+   * The new default in v7 is `corrected`, and this option will be removed in v8.
```
Comment (atscott) [inline packages/router/src/router_module.ts:448]:
```suggestion
   * The new default is `corrected` and this option will be removed in a later version.
```

**Review by atscott — CHANGES_REQUESTED** [review]
We're aiming to merge this change for v11 along with the migration 
 here: https://github.com/angular/angular/pull/38698. 

Could you also update your commit message in the change to add a little more detail and also have a `BREAKING CHANGE` footer note to describe how this might break applications that rely on the default being 'legacy'?

**Comment by atscott** [comment]
Note: rebased PR and removed the note about deprecations, which can be a different PR. We should decide on a plan for how it will get removed rather than just deprecating it with no plan.

**Review by atscott — APPROVED** [review]

**Comment by atscott** [comment]
[presubmit](http://test/OCL:333372924:BASE:333372958:1600896360308:c643ca9f)

**Pull request closed without merging**